src/blueapi/worker/reworker.py

```diff
@@ -68,6 +68,10 @@ def __init__(
     def submit_task(self, name: str, task: Task) -> None:
         active_task = ActiveTask(name, task)
         LOGGER.info(f"Submitting: {active_task}")
+        if self._task_queue.qsize() != 0:
```

Comment:
`qsize()` is approximate and unreliable, per the docs https://docs.python.org/3/library/queue.html#queue.Queue.qsize

The thread-safe solution is something like this
```python
from queue import Queue
from queue import Full
q = Queue(maxsize=1)
try:
        q.put_nowait(1)
        q.put_nowait(2)
    except Full:
        raise Exception("Whatever my exception is")
```

Reply:
thanks, I was wondering about this... will change now.